patches/tModLoader/Terraria/ModLoader/Utilities/HjsonExtensions.cs

```diff
@@ -71,23 +72,42 @@ private static void WriteFancyHjsonValue(TextWriter tw, JsonValue value, int lev
 				}
 
 				if (commentedObject != null) {
+					bool skipFirst = !showBraces;
 					string kwl = GetComments(commentedObject.Comments, "");
+					JsonType? lastJsonType = null;
 
 					foreach (string key in commentedObject.Order.Concat(commentedObject.Keys).Distinct()) {
 						if (!jObject.ContainsKey(key))
 							continue;
 
 						var val = jObject[key];
 
+						if ((val.JsonType != lastJsonType && lastJsonType != null) || lastJsonType == JsonType.Object) {
+							NewLine(tw, 0);
+						}
+
+						if (!string.IsNullOrWhiteSpace(kwl))
+							kwl = (isRootObject ? "" : "\n") + new string('\t', level + (showBraces ? 1 : 0)) + kwl.TrimStart();
+
+						lastJsonType = val.JsonType;
+						
 						tw.Write(kwl);
 
-						NewLine(tw, level + 1);
+						NewLine(tw, level + (showBraces ? 1 : 0));
+
 						kwl = GetComments(commentedObject.Comments, key);
 
+						bool commentedOut = jObject is CommentedWscJsonObject commented && commented.CommentedOut.Contains(key);
+						if (commentedOut)
+							tw.Write("/* ");
```

Comment:
Using block comments to distinguish from user comments?
Can I suggest using `//` (leaving `#` for actual user comments)

Reply:
I had to use block comments because many translations will be multiple lines. This ensures that they are read correctly when loading the mod.